src/components/RoomCard.tsx

```diff
@@ -111,32 +110,32 @@ export const RoomCard: React.FC<{
       >
         <Box gridArea="img" fill>
           <Carousel fill>
-            {room.media?.map((img, i) => (
+            {room?.media?.map((img, i) => (
               <Image key={i} fit="cover" src={img.url} />
             ))}
           </Carousel>
         </Box>
         <Box gridArea="header">
           <Text size="xxlarge" margin={{ bottom: 'xsmall' }}>
-            {room.name}
+            {room?.name}
           </Text>
-          {room.maximumOccupancy && (
+          {room?.maximumOccupancy && (
```

Comment:
I would check if room is defined only once and do not render if not. fyi: optional values is not really good practice in ts. imho for now it does not really matter, we can keep it.

Reply:
Yes I agree. I think the roomcard could be improved in that sense. 

I suggest that the person that touches that component to refactor to MUI and restyle based on the new designs to do these checks. The PR is already really big, so purposvely to pick my battles, and focus only on the essential so that others who are dependent on this PR can proceed.

Reply:
Agreed! Let's tackle this problem in the next PR.